api/src/services/schema.ts

```diff
@@ -0,0 +1,73 @@
+import { Accountability, SchemaOverview } from '@directus/shared/types';
+import { Knex } from 'knex';
+import { flushCaches } from '../cache';
+import getDatabase from '../database';
+import { ForbiddenException, InvalidPayloadException } from '../exceptions';
+import { AbstractServiceOptions, Snapshot, SnapshotDiff } from '../types';
+import { applySnapshot } from '../utils/apply-snapshot';
+import { getSnapshot } from '../utils/get-snapshot';
+import { getSnapshotDiff } from '../utils/get-snapshot-diff';
+
+export class SchemaService {
+	knex: Knex;
+	accountability: Accountability | null;
+	schema: SchemaOverview;
+
+	constructor(options: AbstractServiceOptions) {
+		this.knex = options.knex || getDatabase();
+		this.accountability = options.accountability || null;
+		this.schema = options.schema;
+	}
+
+	async snapshot(): Promise<Snapshot> {
+		if (this.accountability?.admin !== true) throw new ForbiddenException();
+
+		await flushCaches();
+
+		const currentSnapshot = await getSnapshot({ database: this.knex, schema: this.schema });
```

Comment:
There is no need to `flushCache()` as we're fetching a fresh schema that bypasses cache.

```suggestion
		if (this.accountability?.admin !== true) throw new ForbiddenException();

		const currentSnapshot = await getSnapshot({ database: this.knex, schema: this.schema });
```